### PR TITLE
chore: update dependencies and node version

### LINKED ISSUE #4: Update to Node > 12
From GitHub:
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.

`agenthunt/conventional-commit-checker-action@v1.0.0` was flagged as needing updating.

### PR DESCRIPTION
**Pull request by paulschreiber**
chore: update to Node 18,  update to current versions of GitHub actions.

Fixes #4.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,12 +7,12 @@ jobs:
...
+        uses: actions/checkout@v3
 
       - name: Use Node.js
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
-          node-version: '12.x'
+          node-version: 18
       - run: npm install
       - run: npm run build --if-present
```
Comment (agenthunt) [inline .github/workflows/main.yml]: Thanks for the PR. Looks like the check is failing. Seems to be fixed by providing `NODE_OPTIONS=--openssl-legacy-provider` option

```
NODE_OPTIONS=--openssl-legacy-provider npm run build --if-present
```

Comment (paulschreiber) [reply .github/workflows/main.yml]: @agenthunt added you as a collaborator to my fork so you can wrap this up.

Comment (agenthunt) [reply .github/workflows/main.yml]: Thanks. Done and Released v2.0.0

**Comment by paulschreiber** [comment]
@agenthunt should the example in the readme be updated as well? it still has `agenthunt/conventional-commit-checker-action@v1.0.0`

**Pull request merged**